internal/cmd/integration-tests/configs/kafka/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/Shopify/sarama"
+)
+
+const (
+	topicName     = "test_topic"
+	brokerAddress = "kafka:9092"
+)
+
+// This app sends and consumes messages via a Kafka topic.
+
+func main() {
+	go produceMessages()
+	consumeMessages()
+}
+
+func produceMessages() {
+	config := sarama.NewConfig()
+	config.Producer.Return.Successes = true
+
+	var producer sarama.SyncProducer
+	var err error
+
+	for {
+		producer, err = sarama.NewSyncProducer([]string{brokerAddress}, config)
+		if err == nil {
+			break
```

Comment:
What if `NewSyncProducer` never succeeds? Does it retry indefinitely?

Reply:
yeah, I think that's okay because the producer/consumer lifetime is handled by the integration test. The tests have a timeout of 60 seconds and after that, the docker env is brought down. Whether this script retries indefinitely or sends/consumes messages indefinitely does not really matter